sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientJavaDocCodeSnippets.java

```diff
@@ -24,4 +43,892 @@ public TextAnalyticsAsyncClient createTextAnalyticsAsyncClient() {
+                }
+            }
+        });
+        // END: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.detectBatchLanguages#List
+
+        // BEGIN: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.detectBatchLanguagesWithResponse#List-TextAnalyticsRequestOptions
+        List<DetectLanguageInput> detectLanguageInputs1 = Arrays.asList(
+            new DetectLanguageInput("1", "This is written in English.", "US"),
+            new DetectLanguageInput("2", "Este es un document escrito en Español.", "es")
+        );
+
+        // Request options: show statistics and model version
+        final TextAnalyticsRequestOptions requestOptions = new TextAnalyticsRequestOptions().setShowStatistics(true);
+
+        textAnalyticsAsyncClient.detectBatchLanguagesWithResponse(detectLanguageInputs1, requestOptions)
+            .subscribe(response -> {
+                final DocumentResultCollection<DetectLanguageResult> detectedBatchResult = response.getValue();
+                System.out.printf("Model version: %s%n", detectedBatchResult.getModelVersion());
+
+                // Batch statistics
+                final TextDocumentBatchStatistics batchStatistics = detectedBatchResult.getStatistics();
+                System.out.printf("Batch statistics, transaction count: %s, valid document count: %s.%n",
+                    batchStatistics.getTransactionCount(),
+                    batchStatistics.getValidDocumentCount());
+
+                for (DetectLanguageResult detectLanguageResult : detectedBatchResult) {
+                    for (DetectedLanguage detectedLanguage : detectLanguageResult.getDetectedLanguages()) {
+                        System.out.printf("Other detected language: %s, ISO 6391 name: %s, score: %s.%n",
+                            detectedLanguage.getName(),
+                            detectedLanguage.getIso6391Name(),
+                            detectedLanguage.getScore());
+                    }
+                }
+            });
+        // END: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.detectBatchLanguagesWithResponse#List-TextAnalyticsRequestOptions
+    }
+
+    // Entity
+    /**
+     * Code snippet for recognizing entities
+     */
+    public void recognizeEntitiesCodeSnippets() {
+        TextAnalyticsAsyncClient textAnalyticsAsyncClient = createTextAnalyticsAsyncClient();
+
+        // BEGIN: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.recognizeEntities#string
+        String inputText = "Satya Nadella is the CEO of Microsoft";
+        textAnalyticsAsyncClient.recognizeEntities(inputText).subscribe(recognizeEntitiesResult -> {
+            for (NamedEntity entity : recognizeEntitiesResult.getNamedEntities()) {
+                System.out.printf(
+                    "Recognized Named Entity: %s, Type: %s, Subtype: %s, Score: %s.%n",
+                    entity.getText(),
+                    entity.getType(),
+                    entity.getSubtype(),
```

Comment:
```suggestion
                            entity.getSubtype() == null || entity.getSubtype().isEmpty() ? "N/A" : entity.getSubtype(),

```

Reply:
missing  `entity.getLength() `

Reply:
trying to keep the code snippets shorter. Length probably is something that can be skipped here.